lib/beaker/hypervisor/hcloud.rb

```diff
@@ -69,6 +77,25 @@ def create_ssh_key
       hcloud_ssh_key
     end
 
+    # we need to save the date as unix timestamp. Hetzner Cloud labels only support:
+    # "alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between."
+    # https://docs.hetzner.cloud/#labels
+    def vm_deletion_date
+      (DateTime.current + @delete_vm_after.hours).to_i.to_s
+    end
+
+    # check if a server is too old (delete_vm_after is in the past) and delete it
+    # delete it also if it has no delete_vm_after label. All servers are supposed to have that
+    def delete_server_if_required(server)
+      server.destroy if server.labels['delete_vm_after'] && Time.now > Time.at(server.labels['delete_vm_after'].to_i)
+      server.destroy unless server.labels['delete_vm_after']
```

Comment:
How about a more traditional condition:
```suggestion
      if !server.labels['delete_vm_after'] || Time.now > Time.at(server.labels['delete_vm_after'].to_i)
        server.destroy
      end
```